pkg/ops/miningManager.go

```diff
@@ -121,26 +169,167 @@ func (mgr *MiningMgr) Start(ctx context.Context) {
+	rewardTRB := big.NewInt(0).Mul(rewardPerSec, timeDiff)
+
+	singleMinerTip := big.NewInt(0).Div(currentTotalTips, big.NewInt(10)) // Half of the tips are burned(remain in the contract) to reduce inflation.
+	rewardWithTips := big.NewInt(0).Add(singleMinerTip, rewardTRB)
+
+	return mgr.convertTRBtoETH(rewardWithTips)
+}
+
+func (mgr *MiningMgr) convertTRBtoETH(trb *big.Int) (*big.Int, error) {
+	val, err := mgr.database.Get(db.QueriedValuePrefix + strconv.Itoa(tracker.RequestID_TRB_ETH))
+	if err != nil {
+		return nil, errors.New("getting the trb price from the db")
+	}
+	if len(val) == 0 {
+		return nil, errors.New("the db doesn't have the trb price")
+	}
+	priceTRB, err := hexutil.DecodeBig(string(val))
+	if err != nil {
+		return nil, errors.New("decoding trb price from the db")
+	}
+
+	priceTRB = priceTRB.Mul(priceTRB, big.NewInt(tellorCommon.WEI))
+	priceTRB = priceTRB.Div(priceTRB, big.NewInt(tracker.PSRs[tracker.RequestID_TRB_ETH].Granularity()))
+
+	// Big int can't multiple fractions so need to convert the operation into a division.
+	devider := big.NewInt(0).Div(trb, trb)
+	eth := big.NewInt(0).Div(priceTRB, devider)
+	return eth, nil
+}
+
+func (mgr *MiningMgr) txCost() (*big.Int, error) {
+	slotNum, err := mgr.contractGetter.GetUintVar(nil, rpc.Keccak256([]byte("slotProgress")))
+	if err != nil {
+		return nil, errors.Wrap(err, "getting slotProgress")
+	}
+	// This is the price for the last transaction so increment +1
+	// to get the price for next slot transaction.
+	// Slots numbers should be from 0 to 4 so
+	// use mod of 5 in order to save 5 as slot 0.
+	slotNum.Add(slotNum, big.NewInt(1)).Mod(slotNum, big.NewInt(5))
+	dbtxCostName := tellorCommon.PriceTXs + slotNum.String()
+	cost, err := mgr.database.Get(dbtxCostName)
+	if err != nil {
+		return nil, errors.New("getting the tx eth cost from the db")
+	}
+	// No price record in the db yet.
+	if cost == nil {
+		return big.NewInt(0), nil
+	}
+
+	return big.NewInt(0).SetBytes(cost), nil
+}
+
+// saveTXCost calculates the price for a given slot.
+// Since the transaction doesn't include the slot number it was submited for
+// it gets the slot number as soon as the transaction passes and
+// saves it in the database for profit calculations use.
+func (mgr *MiningMgr) saveTXCost(ctx context.Context, tx *types.Transaction) {
+	go func(tx *types.Transaction) {
+		receipt, err := bind.WaitMined(ctx, mgr.ethClient, tx)
+		if err != nil {
+			mgr.log.Error("waiting for transaction completion for  calculating transaction cost err:%v", err)
+		}
+		if receipt.Status != 1 {
+			mgr.log.Error("skiping unsuccessful transaction for calculating transaction cost err:%v", err)
+		}
+		mgr.log.Debug("submit transaction receipt", "txHash:%v", receipt.TxHash)
+		gasUsed := big.NewInt(int64(receipt.GasUsed))
+		txCost := gasUsed.Mul(gasUsed, tx.GasPrice())
+
+		slotNum, err := mgr.contractGetter.GetUintVar(nil, rpc.Keccak256([]byte("slotProgress")))
```

Comment:
add issue for event listener for nonceSubmitted events to determine slot Progress and whether it was yours.

Reply:
updated the issue on the tellor core repo as this needs to be added there.
https://github.com/tellor-io/TellorCore/issues/101